### ISSUE TITLE
on_message_publish username is a table, not string

### ISSUE THREAD

**Issue by saeidakbari** [emqx/emqx#3652]
**Environment**:

- EMQ X version: 4.1.2 
- OS : alpine
- Kernel (e.g. `uname -a`): 
`Linux f18814a69c78 4.19.76-linuxkit #1 SMP Thu Oct 17 19:31:58 UTC 2019 x86_64 Linux`

**What happened and what you expected to happen**:
As what the docs say username for on_message_publish function should be string, but when I debug it using print and some helper functions I realize that its a table without any data in it.
**How to reproduce it (as minimally and precisely as possible)**:
Just run lua-hook example as what the docs say
**Anything else we need to know?**:
 this is my on_message_publish :
```
function on_message_publish(clientid, username, topic, payload, qos, retain)
    print_table(username)
    -- do your job here
    return topic, payload, qos, retain
end
```

**Comment by saeidakbari** [comment]
It seems all `on_message_publish` hooks have the same issue. 
```log
Publish Message(Id=�4�V��˷�, QoS=0, Topic=akbari/home, From=<<"mosq-foCF5legZ47aq69cXj">>, Flags=[], Headers=#{peerhost => {172,19,0,1},
```
if you look at `From` property you will see it just have `clientID` field, username is not exist.

**Comment by HJianBo** [comment]
@saeidakbari Thanks for your feedback. We will fix it at v4.1.3

**Comment by HJianBo** [comment]
Close for duplicated and fixed at v4.1.3
